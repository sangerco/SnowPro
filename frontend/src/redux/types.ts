export interface UserData {
    username: string;
    firstName: string;
    lastName: string;
    email: string;
    avatar: string;
    bio: string;
    videos: string[];
    photos: string[];
    favMountains: string[];
  }


export const FETCH_USER_DATA_REQUEST = 'FETCH_USER_DATA_REQUEST';
export const FETCH_USER_DATA_SUCCESS = 'FETCH_USER_DATA_SUCCESS';
export const FETCH_USER_DATA_FAILURE = 'FETCH_USER_DATA_FAILURE';


export interface FetchUserDataRequestAction {
    type: typeof FETCH_USER_DATA_REQUEST
}

export interface FetchUserDataSuccessAction {
    type: typeof FETCH_USER_DATA_SUCCESS;
    payload: UserData;
}

export interface FetchUserDataFailureAction {
    type: typeof FETCH_USER_DATA_FAILURE;
    payload: string;
}

export type UserActionTypes = 
    FetchUserDataRequestAction
    | FetchUserDataSuccessAction
    | FetchUserDataFailureAction;