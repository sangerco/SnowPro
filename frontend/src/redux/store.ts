import { Action, compose } from "redux";
import { ThunkAction, configureStore } from "@reduxjs/toolkit";
import { logger } from "redux-logger";
import rootReducer from "./reducers/rootReducer";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(logger, composeEnhancers),
});
