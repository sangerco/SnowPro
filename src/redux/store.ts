import { configureStore, combineReducers } from "@reduxjs/toolkit";
import userReducer from "./slices/userSlice";
import authReducer from "./slices/authSlice";
import messageReducer from "./slices/messageSlice";
import messageReplyReducer from "./slices/messageReplySlice";
import reviewReducer from "./slices/reviewSlice";
import reviewReplyReducer from "./slices/reviewReplySlice";
import tagReducer from "./slices/tagSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["messages", "messageReplies", "reviews", "reviewReplies"],
};

const rootReducer = combineReducers({
  users: userReducer,
  auth: authReducer,
  messages: messageReducer,
  messageReplies: messageReplyReducer,
  reviews: reviewReducer,
  reviewReplies: reviewReplyReducer,
  tags: tagReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
