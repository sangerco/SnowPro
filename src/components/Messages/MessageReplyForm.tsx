import React, { useEffect, useState } from "react";
import { useParams } from "react-router";
import { createMessageReply } from "../../redux/slices/messageReplySlice";
import { fetchMessage } from "../../redux/slices/messageSlice";
import { Button, Form } from "semantic-ui-react";
import { RootState, AppDispatch } from "../../redux/store";
import { useDispatch, useSelector } from "react-redux";

const MessageReplyForm: React.FC = () => {
  const { id } = useParams();
  const dispatch = useDispatch<AppDispatch>();
  const messages = useSelector((state: RootState) => state.messages);

  const messageId = id;

  useEffect(() => {
    if (messageId) {
      dispatch(fetchMessage(messageId));
    }
  }, [dispatch, messageId]);

  const message = messages.message;
  const senderId = message ? message.recipientId : "";
  const recipientId = message ? message.senderId : "";
  const subject = message ? `Re: ${message.subject}` : "";
  const body = message ? `>${message.body} \n\n` : "";

  console.log(body);

  const initialMessageReplyState = {
    messageId: messageId ? messageId : "",
    senderId: senderId,
    recipientId: recipientId,
    subject: subject,
    body: body,
  };

  const [formData, setFormData] = useState(initialMessageReplyState);

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    dispatch(createMessageReply(formData));

    setFormData(initialMessageReplyState);
  };

  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const handleBodyChange: React.ChangeEventHandler<HTMLTextAreaElement> = (
    e
  ) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  return (
    <Form onSubmit={handleSubmit}>
      <Form.Field>
        <input
          placeholder="Whatcha talking about?"
          name="subject"
          value={formData.subject}
          onChange={handleChange}
        />
      </Form.Field>
      <Form.TextArea
        placeholder="Message"
        name="body"
        value={formData.body}
        onChange={handleBodyChange}
      />
      <Button type="submit">Send</Button>
    </Form>
  );
};

export default MessageReplyForm;
