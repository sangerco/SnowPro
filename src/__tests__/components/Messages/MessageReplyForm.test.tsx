/* eslint-disable testing-library/no-node-access */
import React from "react";
import MessageReplyForm from "../../../components/Messages/MessageReplyForm";
import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { BrowserRouter } from "react-router-dom";

describe("test Message Reply Form", () => {
  it("should render without crashing", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <MessageReplyForm />
        </BrowserRouter>
      </Provider>
    );
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <MessageReplyForm />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
