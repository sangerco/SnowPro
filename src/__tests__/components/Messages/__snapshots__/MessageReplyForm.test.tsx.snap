// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Message Reply Form should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="ui form"
  >
    <div
      class="field"
    >
      <input
        name="subject"
        placeholder="Whatcha talking about?"
        value=""
      />
    </div>
    <div
      class="field"
    >
      <textarea
        name="body"
        placeholder="Message"
        rows="3"
      />
    </div>
    <button
      class="ui button"
      type="submit"
    >
      Send
    </button>
  </form>
</DocumentFragment>
`;
